backend/amp-live-list.go

```diff
@@ -157,13 +163,56 @@ func createMetadata(r *http.Request) []BlogPosting {
 	return result
 }
 
-func createLiveBlogSample(newStatus int, timestamp time.Time, r *http.Request) LiveBlogSample {
+func createLiveBlogSample(newStatus int, timestamp time.Time, r *http.Request, page Page) LiveBlogSample {
+
+	firstBlogID := r.URL.Query().Get("from")
 	if newStatus > len(blogs) {
 		newStatus = len(blogs)
 	}
 	blogItems := getBlogEntries(newStatus, timestamp)
 	score := createScore(newStatus, 0)
-	return LiveBlogSample{BlogItems: blogItems, FootballScore: score, BlogMetadata: createMetadata(r)}
+	firstItemIndex, _ := getProductIndexFromID(firstBlogID, blogItems)
+	nextPageFirstItemIndex := firstItemIndex + MAX_BLOG_NUMBER
+	nextPageURL := getNextPageUrl(r, page, blogItems, nextPageFirstItemIndex)
+	prevPageURL := getPrevPageUrl(firstItemIndex, r, page, newStatus)
+
+	lenghtCurrentPageBlog := int(math.Min(float64(len(blogItems)), float64(firstItemIndex+MAX_BLOG_NUMBER)))
+
+	return LiveBlogSample{BlogItems: blogItems[firstItemIndex:lenghtCurrentPageBlog],
+		FootballScore: score,
+		BlogMetadata:  createMetadata(r),
+		NextPageURL:   nextPageURL,
+		PrevPageURL:   prevPageURL,
+		PageNumber:    getPageNumberFromProductIndex(firstItemIndex)}
+}
+
+func getNextPageUrl(r *http.Request, page Page, blogItems []BlogItem, nextPageFirstItemIndex int) string {
+	if nextPageFirstItemIndex < len(blogItems) {
+		nextPageFirstItemID := blogs[nextPageFirstItemIndex].ID
+		return buildSourceOrigin(r.Host) + page.Route + "?from=" + nextPageFirstItemID
```

Comment:
you build the same URL in both methods getNextPageUrl and getPrevPageUrl. Make this method only return the new index and build the URL somewhere else.